File: B-happy-array/sol.py
def solve():
    n = int(input())
    a = list(map(int, input().split()))
    
    # evaluate each index separately
    # find maximum minimum operations needed
    mxmn = 0
 
    for i in range(1, n+1):
        mxmn = max(mxmn, (a[i-1] * (-1)) // i + 1)
 
    print(mxmn)
 
if __name__ == "__main__":
    solve()

File: B-happy-array/sol.cpp
#include<bits/stdc++.h>
#pragma optimize("O3,unroll-loops")
using namespace std;
 
#define int long long int
 
auto solve() {
        int n;cin>>n;
 
        // evaluate each index separately, find maximum minimum operations needed
        int mxmn = 0;
 
        for (int i=1;i<=n;i++) {
                int ai;cin>>ai;ai*=-1;
                mxmn = max(mxmn, ai/i+1);
        }
 
        cout << mxmn << "\n";
}
 
int32_t main() {
        ios_base::sync_with_stdio(false);
        cin.tie(0);
        solve();
}
